dev/src/codewind/connection/UserProjectCreator.ts

```diff
@@ -120,7 +120,35 @@ namespace UserProjectCreator {
             projectTypeInfo = userProjectType;
         }
 
-        await requestBind(connection, projectName, pathToBind, projectTypeInfo.language, projectTypeInfo.projectType);
+        return bind(connection, projectName, pathToBind, projectTypeInfo);
+    }
+
+    async function bind(connection: Connection, projectName: string,
+                        pathToBind: string, projectTypeInfo: IProjectTypeInfo):
+                        Promise<INewProjectInfo | undefined> {
+        if (connection.remote) {
+            return bindRemote(connection, projectName, pathToBind, projectTypeInfo);
+        } else {
+            return bindLocal(connection, projectName, pathToBind, projectTypeInfo);
+        }
+    }
+
+    async function bindLocal(connection: Connection, projectName: string,
+                             pathToBind: string, projectTypeInfo: IProjectTypeInfo):
+                             Promise<INewProjectInfo | undefined> {
+
+        await requestLocalBind(connection, projectName, pathToBind, projectTypeInfo.language, projectTypeInfo.projectType);
+        return { projectName, projectPath: pathToBind };
```

Comment:
The function header says this can return `undefined` but I don't see a code path for that